### DISCUSSION TITLE
publish patched version

### DISCUSSION THREAD

**Discussion by flefebure** [verdaccio/verdaccio#2177] [Q&A]
Hi,

I'm new to verdaccio (and npm in extent)
I want to patch a public package (highcharts-8.1.2)

So :

- I've installed verdaccio and verfied npm install was ok from workstation
- I've installed locally the package
- patched it locally
- changed the version number in packag.json "8.1.2" -> "8.1.2-SOFTBRIDGE-1.0.0"
- run "npm publish" on the vercciano registry

In the verdaccio log I see :
```
http --- 192.168.50.1 requested 'PUT /highcharts'
 info --- flefebure is allowed publish for highcharts
 info --- flefebure allowed unpublish for highcharts
 info --- unpublishing highcharts@0.0.1
 info --- unpublishing highcharts@0.0.2
 info --- unpublishing highcharts@0.0.3
 info --- unpublishing highcharts@0.0.4
 info --- unpublishing highcharts@0.0.5
```
... etc etc with all the highcharts version
Why theses unpublishes ?

Now I can do a "npm install highcharts-8.1.2-SOFTBRIDGE-1.0.0" through verdaccio
But I can't do no more a "npm install highcharts-8.1.2" 

I don't want to lose the abality to fetch official versions of the package.

Cant you explain ?
Thanks
Franck

**Comment by flefebure** [comment]
Note that I also tried this best practice :

> You want to temporarily use your version, but return to the public one as soon as it's updated.
> 
> In order to avoid version conflicts, you should use a custom pre-release suffix of the next patch version. For example, if a public package has version 0.1.2, you can upload 0.1.3-my-temp-fix.
> 
>  npm version 0.1.3-my-temp-fix
>  npm publish --tag fix --registry http://localhost:4873

But this procedure also locks all others version of the package

**Comment by DanielRuf** [comment]
In general I would recommend to use the package `patch-package` for this instead of forking a package.

**Reply by flefebure** [reply]
I work with Vaadin, the project webpack/node config files are generated dynamically 
- with Maven/Gradle in "production" mode
- with the webpack dev server in "dev" mode

So the patching of a dependancy on-the-fly seemed to me more complicated that have a forked module in a local NPM proxy.
But I could'nt figure it was complicated/(impossible?) to override a package version in the proxy

Looks like Verdaccio doesn't fit here, I will search another solution
